File: binary_to_decimal_converter.R
x<- as.character(readline("Introduce el numero binario para convertirlo en decimal:"))
y<-0
for (i in (nchar(x)-1):0) {
  k<- nchar(x)-i
  y<- y+as.numeric(substr(x,k,k))*(2^i)}
cat("Tu número en base decimal es",y)

File: binary_to_decimal_converter.py
# ------------------------------------------------------------------------
#
#
# Made with python 3.8.8
#
#
# ------------------------------------------------------------------------

def bdecimal():
    x= input("Introduce el numero binario para convertirlo en decimal: ")
    y= 0
    posicion = len(x)-1
    for i in x:
        k = posicion
        y = y + (int(i)*2**k)
        posicion=posicion-1

    return("Tu número en base decimal es "+str(y))
